### DISCUSSION TITLE
ecore-glsp User Experience?

### DISCUSSION THREAD

**Discussion by GaryKopp** [eclipse-emfcloud/emfcloud#87] [All other]
Rather than mess with Docker, I built ecore-glsp the long-winded way from the command line. On starting it and going to port 3000 in Chrome, I'm wondering if I'm seeing the application the way it was intended, or if I'm missing something. Theia opens looking pretty empty. The explorer view is not visible, and the editor area is empty. The only clue that I'm looking at the ecore-glsp application is a specialized menu bar, whose help indicates it's coming from browser-app. It looks like the user experience using the application is expected to be, and *please* correct me if I'm wrong, to open the explorer view, navigate to some .ecore file (such as ecore.ecore in client/workspace/ecore/model) and open it. And then things like the tree view are accessible by right-clicking on the .ecore file in the explorer. Is that the intended usage of the example?

**Comment by planger** [comment]
As Ecore GLSP is about the Ecore editor and not a polished and readily bundled application, it doesn't add any customizations or extensions of the Theia workbench (ie. the application around the editor) itself. So it doesn't open specific views, like the explorer view, or a getting started page or so. This is rather something which applications that integrate Ecore GLSP would do. So when opening a plain Theia with Ecore GLSP (as you did), you'll see the default initial state of Eclipse Theia workbench being opened on the `workspace` folder at https://github.com/eclipse-emfcloud/ecore-glsp/tree/master/client/workspace

The workspace folder contains a couple of ecore files for testing, among them ecore.ecore. You should be able to open the Explorer view via the menu bar (e.g. View -> Explorer) right-click on any ecore file there and open it (Open With -> Ecore Editor or so).
The default workspace that is being opened is defined in the npm scripts here: https://github.com/eclipse-emfcloud/ecore-glsp/blob/f7d3c343b7c6f9d57c1dbdfc3c8fb673b2395fdc/client/browser-app/package.json#L30

**Reply by GaryKopp** [reply]
> The workspace folder contains a couple of ecore files for testing, among them ecore.ecore.

Even though ecore.ecore is provided for "testing," is it not also quite special in the sense that ecore-glsp uses it internally as the "type schema" for any instance model opened in the application? Thanks for your help!

**Reply by planger** [reply]
Exactly it is not special in that sense, just yet another ecore model in this context.

**Reply by GaryKopp** [reply]
What I found a bit confusing or even misleading as I tried to understand this is that ecore.ecore is present in the the workspace as it should be as an editable test case, but that very same file is accessed directly by ecore-glsp as the application's type schema. So it's serving a dual purpose in the deployment. But at least now I believe I understand the usages.
